src/run_autodiscovery.py

```diff
@@ -0,0 +1,237 @@
+   # . -> _
+   # / -> .
+   metric_name.replace(r'\.','_')
+   metric_name.replace(r'/','.')
+   return metric_name
+
+async def init_session():
+    async with init_gcp_client_session() as gcp_session:
+        token = await create_token(logging_context, gcp_session)
+        if token:
+            await get_metric_descriptors(gcp_session,token)
+        else:
+            logging_context.log('Unable to acquire authorization token.')
+
+
+async def _sent_metadata_to_dt(dynatrace_url: Text,dynatrace_access_key, dt_session, metric_metadata_list):
+    data_input = "\n".join([metadata.to_json() for metadata in metric_metadata_list])
+
+    dt_url = f"{dynatrace_url.rstrip('/')}/api/v2/settings/objects"
+    response = await dt_session.post(
+        url=dt_url,
+        headers={
+            "Authorization": f"Api-Token {dynatrace_access_key}",
+            "Content-Type": "application/json"
+        },
+        data=data_input,
+        verify_ssl=get_should_require_valid_certificate()
+    )
+
+    if response.status != 200:
+        print("Error During sending metadata meterics")
+        return False
+    else:
+        return True
+
+
+async def get_metric_descriptors(gcp_session,token):
+    headers = {
+    "Accept": "application/json",
+    "Authorization": f"Bearer {token}"}
+    url = f"https://monitoring.googleapis.com/v3/projects/{projectID}/metricDescriptors"
+    resp = await gcp_session.request('GET', url=url, headers=headers)
+    page = await resp.json()
+
+    metric_yaml = [
+        {
+            "value": x.get("type"),
+            "key":  "cloud.gcp." + metric_name_parse(x.get("type"),type_metrickind_caster(x.get("metricKind"),x.get("valueType"))),
+            "displayName":  x.get("displayName"),
+            "name": x.get("displayName"),
+            "description": x.get("description"),
+            "type": type_metrickind_caster(x.get("metricKind"),x.get("valueType")),
+            "gcpOptions":
+                {
+                    "ingestDelay":  int(x.get("metadata",{}).get("ingestDelay","60s")[:-1]),
+                    "samplePeriod": int(x.get("metadata",{}).get("samplePeriod","60s")[:-1]),
+                    "valueType":    x.get("valueType"),
+                    "metricKind":   x.get("metricKind"),
+                    "unit":         unit_caster(x.get("unit"))
+            },
+            "dimensions": [ 
+                {
+                "key":      dimension.get("key"), 
+                "value":    "label:metric.labels." + dimension.get("key"),
+                } for dimension in x.get("labels") or []],
+            "monitored_resources_types": x.get("monitoredResourceTypes",[])
+        } for x in page["metricDescriptors"]]
+    
+    
+
+
+
+    metric_list = [(Metric(**x), x.get("monitored_resources_types"),x) for x in metric_yaml if x.get("gcpOptions").get("valueType").upper() != "STRING" ]
+    discovered_resource_type_list = list(filter(lambda x: discovered_resource_type in x[1], metric_list))
```

Comment:
So you're like fetching all metric descriptors and then throwing most of them out to only have for function, right? 
1. How big is the response, how many metric descriptors, is there paging? 
2. Is this possible to add some filter to query so that you only fetch descriptors for lambda?